server/app/controllers/AddTask.scala

```diff
@@ -73,17 +82,31 @@ class AddTask @Inject()(executor: DynamicSuiteExecutor, dao: TaskDao, val messag
         }
 
         val checkTrait = Try(findTraitName(f.suite)).toFuture
-        //TODO: split error messages: 1) trait check; 2) cannot be saved
-        checkTrait.flatMap { traitName =>
-          addTaskIfValid(traitName)
-        }.recover {
+
+        (for {
+          traitName <- checkTrait errorMsg addTaskErrorOnSolutionTrait
+          result <- addTaskIfValid(traitName) errorMsg cannotAddTaskToDatabase
+        } yield result).recover {
           case NonFatal(e) => BadRequest {
-            addTaskViewWithError(addTaskErrorOnSolutionTrait, "Task cannot be saved", Some(e))
+            addTaskViewWithError(e.getMessage, ex = Some(e))
           }
         }
       }
     )
   }
+
+  def getSolutionTemplate: WebSocket = WebSocket.accept { req =>
+    def getTemplate(jsValue: JsValue) = Json.toJson(
+      SolutionTemplate(
+        Try(CodeParser.getSolutionTemplate((jsValue \ "solution").as[String]))
+          .toOption
+          .getOrElse("can't parse code")
```

Comment:
I think nothing else is needed to be sent. I just wrapped string into SolutionTemplate type to have some type-safety